File: backend/src/domain/entities/Event.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('events')
export class Event {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ name: 'creator_member_id' })
  creatorMemberId: number;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column({ type: 'varchar', name: 'banner_url', nullable: true })
  bannerUrl: string;

  @Column({ type: 'varchar', nullable: true })
  location: string;

  @Column({ type: 'timestamp', name: 'start_time', nullable: true })
  startTime: Date;

  @Column({ type: 'timestamp', name: 'end_time', nullable: true })
  endTime: Date;

  @Column({ type: 'timestamp', name: 'registration_start_at', nullable: true })
  registrationStartAt: Date;

  @Column({ type: 'timestamp', name: 'registration_end_at', nullable: true })
  registrationEndAt: Date;

  @Column({ type: 'integer', name: 'max_capacity', nullable: true })
  maxCapacity: number;

  @Column({ type: 'integer', name: 'interested_count', default: 0 })
  interestedCount: number;

  @Column({ type: 'boolean', name: 'is_published', default: false })
  isPublished: boolean;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/eventmodule/domain/entity/Event.java
package com.service.backend.eventmodule.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("events")
public class Event {

    @Id
    private Long id;

    @Column("organization_id")
    private Long organizationId;

    @Column("creator_member_id")
    private Long creatorMemberId;

    private String title;

    private String description;

    @Column("banner_url")
    private String bannerUrl;

    private String location;

    @Column("start_time")
    private LocalDateTime startTime;

    @Column("end_time")
    private LocalDateTime endTime;

    @Column("registration_start_at")
    private LocalDateTime registrationStartAt;

    @Column("registration_end_at")
    private LocalDateTime registrationEndAt;

    @Column("max_capacity")
    private Integer maxCapacity;

    @Column("interested_count")
    @Builder.Default
    private Integer interestedCount = 0;

    @Column("is_published")
    @Builder.Default
    private Boolean isPublished = false;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
